rabbit-escape-ui-swing/src/rabbitescape/ui/swing/GameUi.java

```diff
@@ -425,6 +427,17 @@ public void actionPerformed( ActionEvent evt )
         } );
 
         MenuTools.clickOnKey( menu.pause, "pause", KeyEvent.VK_P );
+        
+        menu.speed.addActionListener( new ActionListener() 
+        {
+            @Override
+            public void actionPerformed( ActionEvent evt )
+            {
+                gameLaunch.setSpeed( speeds[speedIndex = -1 * ( speedIndex -1 )] );
```

Comment:
Mad skillz. I'd prefer the modification of speedIndex be on the line before. Also, if you're going to be that clever I think it deserves a comment.